### DISCUSSION TITLE
doc feedback - Guidance on Optimistic Currency strategies

### DISCUSSION THREAD

**Discussion by nicholas-brooks** [JasperFx/marten#2798] [General]
I was looking to see how to add optimistic concurrency to a document.  Was about to use `UseOptimisticConcurrency` but needed the version value sent from the client application and I was trying to figure out how `UseOptimisticConcurrency` works to get the version.  Lo and behold, there is IVersion which is for this particular use-case.

I note that there is brief discussion about when to use what strategy at the end of each of `UseOptimisticConcurrency` and "Use IVersioned" but I would've found it easier if it was put into its own section. e.g.

```
# Optimistic Concurrency

## UseOptimisticConcurrency (Implicit?)

## IVersion (Explicit?)

## Which should I use?
```

**Comment by mysticmind** [comment]
@nicholas-brooks We will re-organize the docs a bit to make it clear. Thanks for the feedback.

**Comment by mysticmind** [comment]
Moving this to discussions, will bring back to issues when we update docs.
